### ISSUE TITLE
Unhandled promise rejections will not be caught by Sentry warning not working

### ISSUE THREAD

**Issue by jer-sen** [getsentry/sentry-react-native#2268]
### Environment

How do you use Sentry?
Sentry SaaS (sentry.io)

Which SDK and version?
@sentry/react-native@3.4.3

### Steps to Reproduce

Run the app.

### Expected Result

There should be one of these logs from https://github.com/getsentry/sentry-react-native/blob/f916f2b3afbb9884f9037b8486539ca6cc8b327c/src/js/integrations/reactnativeerrorhandlers.ts#L141:

- `Unhandled promise rejections will not be caught by Sentry. Read about how to fix this on our troubleshooting page.`
- `Unhandled promise rejections will be caught by Sentry.`
- `Unhandled promise rejections will not be caught by Sentry. Read about how to fix this on our troubleshooting page.`

### Actual Result

No message in logs.

Replacing:

`logger.warn("Unhandled promise rejections will not be caught by Sentry. Read about how to fix this on our troubleshooting page.");`

by

`setTimeout(() => console.warn("Unhandled promise rejections will not be caught by Sentry. Read about how to fix this on our troubleshooting page."), 0);`

solves the issue (`setTimeout` and `console` instead of `logger` are both required). Maybe the logger is not yet ready at this time...

**Comment by marandaneto** [comment]
@jer-sen where do you init the Sentry SDK? I can't reproduce that, I always get the logs.
Can you provide a minimal reproduction example? thanks.

**Comment by marandaneto** [comment]
@jer-sen so it's not about the unhandled promise not working at all, but rather the logging that tells you if it's working or not, does it make sense?

**Comment by jer-sen** [comment]
Still an issue!

**Comment by krystofwoldrich** [comment]
In our `[sample](https://github.com/getsentry/sentry-react-native/blob/396f4f7b5e41360fd2bc089feda98009d721ba6c/sample/src/App.tsx#L26)` app we also init Sentry before `AppRegistry.registerComponent(appName, () => App);` but the JS logs are printed out as expected.

Example from a run on iOS Sentry SDK version `4.7.1`.
```
2022-10-27 15:00:17.510197+0200 sample[18388:6697968] [javascript] 'Sentry Logger [log]:', 'Unhandled promise rejections will be caught by Sentry.'
```

So is this still an issue @jer-sen ?

**Comment by jer-sen** [comment]
I will update and try but not now. Probably in 2 weeks.

**Comment by marandaneto** [comment]
any updates here  @jer-sen ?

**Comment by jer-sen** [comment]
> any updates here @jer-sen ?

Not yet, but still in my mind

**Comment by krystofwoldrich** [comment]
Hi @jer-sen, any update?

**Comment by jer-sen** [comment]
Not yet, and not in the coming days. Maybe you should close the issue and I will update you when I will have time to check.

**Comment by AndrewTotsky** [comment]
@jer-senWe had the same problems. Solution turned out by chance. 

Amplitude didn’t send events about app’s start. It happened because we tried to generate user_id using uuid depending library. Uuid depending library requires react-native-get-random-values. Import of depending library was happening too lately (read more here https://github.com/uuidjs/uuid#getrandomvalues-not-supported), that’s why generated identifier was incorrect. 

In conclusion, we started getting events and saw them in Sentry only after adding import 'react-native-get-random-values'; in index.js -- 

Hope it’ll help you.

**Comment by AndrewTotsky** [comment]
@jer-sen 
We had the same problems. Solution turned out by chance. 

Amplitude didn’t send events about app’s start. It happened because we tried to generate user_id using uuid library. Uuid depending library requires react-native-get-random-values (read more here https://github.com/uuidjs/uuid#getrandomvalues-not-supported). Import of depending library was happening too lately, that’s why generated identifier was incorrect. 

In conclusion, we started getting events and saw them in Sentry only after adding import 'react-native-get-random-values'; in index.js

Hope it’ll help you.

**Comment by jer-sen** [comment]
@krystofwoldrich I have (finally!) updated to @sentry/react-native@4.15.2 (latest version supported by sentry-expo) and the issue is still here. When I patch (with patch-package) @sentry/react-native like this
```
diff --git a/node_modules/@sentry/react-native/dist/js/integrations/reactnativeerrorhandlers.js b/node_modules/@sentry/react-native/dist/js/integrations/reactnativeerrorhandlers.js
index ab21a2a..09388e5 100644
--- a/node_modules/@sentry/react-native/dist/js/integrations/reactnativeerrorhandlers.js
+++ b/node_modules/@sentry/react-native/dist/js/integrations/reactnativeerrorhandlers.js
@@ -93,14 +93,20 @@ export class ReactNativeErrorHandlers {
             const Promise = require('promise/setimmediate/es6-extensions');
             if (Promise !== RN_GLOBAL_OBJ.Promise) {
                 logger.warn('Unhandled promise rejections will not be caught by Sentry. Read about how to fix this on our troubleshooting page.');
+                // For message to appear
+                setTimeout(() => console.warn('Console.warn: Unhandled promise rejections will not be caught by Sentry. Read about how to fix this on our troubleshooting page.'), 0);
             }
             else {
                 logger.log('Unhandled promise rejections will be caught by Sentry.');
+                // For message to appear
+                setTimeout(() => console.log('Console.log: Unhandled promise rejections will be caught by Sentry.'), 0);
             }
         }
         catch (e) {
             // Do Nothing
             logger.warn('Unhandled promise rejections will not be caught by Sentry. Read about how to fix this on our troubleshooting page.');
+            // For message to appear
+            setTimeout(() => console.warn('Console.warn: Unhandled promise rejections will not be caught by Sentry. Read about how to fix this on our troubleshooting page.'), 0);
         }
     }
     /**

```

I can only see `Console.log: Unhandled promise rejections will be caught by Sentry.` in event breadcrumbs.

**Comment by krystofwoldrich** [comment]
@jer-sen This is likely caused by the order in which the integrations are initialized. The `Breadcrumbs` integration is initialized after the `ReactNativeErrorHandlers `.

**Comment by jer-sen** [comment]
@krystofwoldrich ok but then in my opinion the warning is useless if we can't get it in the breadcrumbs

**Comment by krystofwoldrich** [comment]
@jer-sen It's meant for local debugging once it's set up it should not change in production.

**Comment by jer-sen** [comment]
Ok. With dev client build (using Expo) I can see the logs.
